File: week-3/W3.js
function main() {
    let src = "https://padax.github.io/taipei-day-trip-resources/taipei-attractions-assignment.json";

    fetch(src)
    .then(function(response) {
        return response.json();
    })
    .then(function(result) {
        let content = document.createElement('div');
        document.body.appendChild(content);
        content.setAttribute("class", "content");
    
        Array.from(Array(8)).forEach((_, index) => {
            /**
             * all section
             */
            const all = document.createElement('div');
            content.appendChild(all);
            all.setAttribute("class", "all");
        
            /**
             * img section
             */
            const myString = result.result.results[index].file;
            const splits = myString.split(['https']);
            const src = `https${splits[1]}`;
            const img = document.createElement('img');
            all.appendChild(img);
            img.setAttribute("src", src);
            img.setAttribute("class", "img");
        
            /**
             * name section
             */
            const name = document.createElement('div');
            all.appendChild(name);
            name.textContent = result.result.results[index].stitle;
            name.setAttribute("class", "img_name");
        });
    });
};

main();

File: week-3/W3.py
import urllib.request as request
import json
import csv
src="https://padax.github.io/taipei-day-trip-resources/taipei-attractions-assignment.json"
with request.urlopen(src) as response:
    data=json.load(response)

f = open('data.csv', 'w', encoding='utf-8')
w = csv.writer(f)

title = ['景點名稱','區域','經度','緯度','第一張圖檔網址']
w.writerow(title)
    
for test in data['result']['results']:
    total_data = []
    total_data.append(test['stitle'])
    total_data.append(test['address'][5]+test['address'][6]+test['address'][7])
    total_data.append(test['longitude'])
    total_data.append(test['latitude'])
    total_data.append('https' + test['file'].split('https')[1])
    
    w.writerow(total_data)

f.close()